packages/contracts-core/contracts/test/Home.t.sol

```diff
@@ -115,4 +114,78 @@ contract HomeTest is NomadTestWithUpdaterManager {
         home.update(oldRoot, newRoot, sig);
         assertEq(newRoot, home.committedRoot());
     }
+
+    function test_updateRejectFailedState() public {
+        bytes memory messageBody = "";
+        uint32 destinationDomain = remoteDomain;
+        bytes32 recipient = bytes32(uint256(uint160(vm.addr(1505))));
+        home.dispatch(destinationDomain, recipient, messageBody);
+        test_improperUpdate();
+        bytes32 newRoot = home.root();
+        bytes32 oldRoot = home.committedRoot();
+        bytes memory sig = signHomeUpdate(updaterPK, oldRoot, newRoot);
+        vm.expectRevert("failed state");
+        home.update(oldRoot, newRoot, sig);
+    }
+
+    function test_suggestUpdate() public {
+        bytes32 committedRoot = home.committedRoot();
```

Comment:
consider: refactor sending a message into its own helper function so this functionality isn't always repeated (makes someone have to re-review every line of code carefully each time)